fiaas_skipper/deploy/bootstrap.py

```diff
@@ -4,37 +4,51 @@
 
 import logging
 
+from k8s.client import NotFound
 from k8s.models.common import ObjectMeta
-from k8s.models.job import JobSpec, Job
-from k8s.models.pod import Container, PodSpec, PodTemplateSpec
+from k8s.models.pod import Container, PodSpec, Pod
 from prometheus_client import Counter
 
+BOOTSTRAP_POD_NAME = "fiaas-deploy-daemon-bootstrap"
+
 LOG = logging.getLogger(__name__)
 
 bootstrap_counter = Counter("bootstraps_triggered", "A deployment caused a bootstrap to be triggered")
 
 
-class Bootstrapper(object):
+class BarePodBootstrapper(object):
     def __init__(self, cmd_args=()):
         self._cmd_args = cmd_args
 
     def __call__(self, deployment_config, channel, spec_config=None):
         bootstrap_counter.inc()
         LOG.info("Bootstrapping %s in %s", deployment_config.name, deployment_config.namespace)
-        object_meta = ObjectMeta(generateName="bootstrap-fiaas-",
-                                 namespace=deployment_config.namespace)
-        container = Container(
-            name="fiaas-deploy-daemon-bootstrap",
-            image=channel.metadata['image'],
-            command=["fiaas-deploy-daemon-bootstrap"] + self._cmd_args
-        )
-        pod_spec = PodSpec(containers=[container], serviceAccountName="default", restartPolicy="Never")
-        pod_annotations = _get_pod_annotations(spec_config)
-        pod_metadata = ObjectMeta(name="fiaas-deploy-daemon-bootstrap", annotations=pod_annotations)
-        pod_template_spec = PodTemplateSpec(metadata=pod_metadata, spec=pod_spec)
-        job_spec = JobSpec(template=pod_template_spec)
-        job = Job(metadata=object_meta, spec=job_spec)
-        job.save()
+        try:
+            Pod.delete(name=BOOTSTRAP_POD_NAME, namespace=deployment_config.namespace)
+        except NotFound:
+            pass
+        pod_spec = _create_pod_spec(self._cmd_args, channel)
+        pod_metadata = _create_pod_metadata(deployment_config, spec_config)
+        pod = Pod.get_or_create(metadata=pod_metadata, spec=pod_spec)
+        pod.save()
+
+
+def _create_pod_spec(args, channel):
+    container = Container(
+        name="fiaas-deploy-daemon-bootstrap",
+        image=channel.metadata['image'],
+        command=["fiaas-deploy-daemon-bootstrap"] + args
+    )
+    pod_spec = PodSpec(containers=[container], serviceAccountName="default", restartPolicy="Never")
+    return pod_spec
+
+
+def _create_pod_metadata(deployment_config, spec_config):
+    pod_annotations = _get_pod_annotations(spec_config)
+    pod_metadata = ObjectMeta(name=BOOTSTRAP_POD_NAME,
```

Comment:
I think we should add a label that can be used to select all bootstrapping pods in a namespace or cluster, which could be useful for troubleshooting. I'm not sure if it should live here or in the fiaas.yml we are using. Thoughts?

Reply:
The fiaas.yml isn't applied here, so it must be in this code. And you're right, we should add a label. I'll fix.